build/lite/docker/pull.go

```diff
@@ -60,27 +63,38 @@ func (d Docker) ImagesPullByList(images []string) {
 }
 
 func (d Docker) ImagePull(image string) error {
+	var (
+		named      reference.Named
+		err        error
+		cli        *client.Client
+		authConfig types.AuthConfig
+		out        io.ReadCloser
+	)
+	named, err = reference.ParseToNamed(image)
+	if err != nil {
+		logger.Warn("image information parsing failed: %v", err)
+		return err
+	}
 	var ImagePullOptions types.ImagePullOptions
 	ctx := context.Background()
-	cli, err := client.NewClientWithOpts(client.FromEnv, client.WithAPIVersionNegotiation())
+	cli, err = client.NewClientWithOpts(client.FromEnv, client.WithAPIVersionNegotiation())
 	if err != nil {
+		logger.Warn("docker client creation failed: %v", err)
 		return err
 	}
-	if d.Username != "" && d.Password != "" {
-		authConfig := types.AuthConfig{
-			Username: d.Username,
-			Password: d.Password,
-		}
+	authConfig, err = utils.GetDockerAuthInfoFromDocker(named.Domain())
+	if err != nil {
```

Comment:
err != nil ?

Reply:
I think we can tolerate the fault here, because we could mount config into registry.

Reply:
OK